src/query/exist_query.rs

```diff
@@ -0,0 +1,287 @@
+use core::fmt::Debug;
+
+use columnar::{ColumnType, DynamicColumn};
+
+use super::range_query::VecCursor;
+use super::ConstScorer;
+use crate::core::SegmentReader;
+use crate::docset::{DocSet, TERMINATED};
+use crate::query::explanation::does_not_match;
+use crate::query::{EnableScoring, Explanation, Query, Scorer, Weight};
+use crate::schema::FieldType;
+use crate::{DocId, Score, TantivyError};
+
+/// Query that matches all of the documents.
+///
+/// All of the document get the score 1.0.
+#[derive(Clone, Debug)]
+pub struct ExistsQuery {
+    field: String,
```

Comment:
Renaming could make things a little bit less confusing with the `Field` type
```suggestion
    field_name: String,
```